test/integration/transaction.test.js

```diff
@@ -403,6 +403,50 @@ if (current.dialect.supports.transactions) {
           });
         });
 
+        it('supports for update with skip locked', function() {
+          const User = this.sequelize.define('user', {
+              username: Support.Sequelize.STRING,
+              awesome: Support.Sequelize.BOOLEAN
+            }),
+            self = this;
+
+          return this.sequelize.sync({ force: true }).then(() => {
+            return Promise.all([
+              User.create(
+                { username: 'jan'}
+              ),
+              User.create(
+                { username: 'joe'}
+              )
+            ]);
+          }).then(() => {
+            return self.sequelize.transaction().then(t1 => {
+              return User.findAll({
+                limit: 1,
+                lock: true,
+                skipLocked: true,
```

Comment:
In first transaction there is no need to skip locks, I assume all rows are free at this time.